src/pages/dataTableUtilities/constants.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+export const COLUMN_TYPES = {
+  CHECKABLE: 'checkable',
+  DATE_EDITABLE: 'editableDate',
+  DATE: 'date',
+  DROP_DOWN: 'dropdown',
+  ICON: 'icon',
+  NUMERIC_EDITABLE: 'editableNumeric',
+  NUMERIC: 'numeric',
+  STRING_EDITABLE: 'editableString',
+  STRING: 'string',
+};
+
+export const COLUMN_KEYS = {
+  AVAILABLE_QUANTITY: 'availableQuantity',
+  BATCHES: 'batches',
+  BATCH: 'batch',
```

Comment:
I don't see this being used

Reply:
im stupid - it's batches that isn't used. I think this seems a little confused? There are keys in here mixed with fields of the `COLUMNS` object? I think either have just the keys, or separate them into two different objects? Keys are fields/getters on realm models/objects - it seems like this could actually introduce more errors - using a key which is actually a field in `COLUMNS`?

Reply:
Yeah, it's a bit confusing to differentiate how different strings as used here because they get "post-processed" at other places in the code base. Updated now, let me know if I'm still not understanding it.